### PR TITLE
[WIP] Switch to new Thunder UX

### PR DESCRIPTION
**Pull request by mtodor**
Make sure these boxes are checked before submitting your pull request - thank you!

- [x] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [x] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/8.x-1.x/docs/development.md#how-to-run-the-tests))
- [x] Necessary update hooks are provided.
- [x] User roles have correct access for new introduced permission.
- [x] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [x] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

**Comment by mtodor** [comment]
**Still WIP** just creating PR for easier testing and tracking of changes.

File: modules/thunder_updater/thunder_updater.drush.inc [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @file
+ * Contains the code to generate the custom drush commands for module.
```
Comment (dbosen) [inline modules/thunder_updater/thunder_updater.drush.inc]: Cleanup copy and pasted texts :)

File: modules/thunder_updater/updates.yml [hunk 2]
```diff
@@ -57,3 +57,7 @@ v1.1:
         '#title': 'Use new riddle media integration'
         '#description': "<p>We want to use the new riddle media integration. It's much more user-friendly than the old way. You can select riddles from the entity browser now</p>"
         '#description_failed': '<p><strong>There was a problem during the update.</strong> We were not able to convert you current riddles into new media_entity based ones.<p>'
+    v1_1__thunder_admin_ux:
+        '#title': 'New Thunder Admin UX'
+        '#description': "<p>Thunder now comes with new improved administration user interface.</p>"
+        '#description_failed': '<p><strong>There was a problem during the update.</strong> New Thunder Admin UX require plenty new configurations to be adjusted and imported in order to use it properly. We suggest you to take a look at execution of update hook for enabling new administration theme and manually adjust missing configurations.<p>'
```
Comment (dbosen) [inline modules/thunder_updater/updates.yml]: Add information to check logs for update problems.

**Comment by mtodor** [comment]
Additional refactoring:
- Logger: clear log on output (there is no more call to clearLog in update hooks) and add one function that will handle drush and update.php output.
- Updater: inject Logger service and expose it over logger() function.
- ConfigHandler: add handling of deseralization of yml files and also handling of saving/reading of .../config/update files.
- Updater: inject ConfigHandler service to have functionality of fetching update definition from files
- Updater: add import config functionality (similar to install modules)

At end update hook should do: get updater service, run import configs function, run execute updates method and return output of logger (updater->logger()->output()).

**Comment by mtodor** [comment]
All review comments are addressed, except removing of branch testing in `.travis.yml`. And that should stay till the end, so that we have screenshots upload in case of test failures.

File: composer.json [hunk 3]
```diff
@@ -66,6 +66,15 @@
             },
             "drupal/amptheme": {
                 "Don't use network for loading logo information": "https://www.drupal.org/files/issues/amptheme-do_not_use_the_network_for_loading_logo_information-2753089-7.patch"
+            },
+            "drupal/entity_browser": {
+                "Provide inline entity form FieldWidgetDisplay": "https://www.drupal.org/files/issues/2858438_3.patch"
```
Comment (dbosen) [inline composer.json]: This patch contains a var_dump, please remove this. The maintainer also needs other changes, fix those issues as well, to get that merged. See: https://www.drupal.org/node/2858438

**Pull request merged**